packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,168 @@
+// NOTE: Breakpoints can't come from the theme because we need access to them at
+// initialization time and media-queries can't use CSS variables. For users with
+// custom breakpoints, we might want to provide a way to configure them globally
+// instead of through the theme.
+const breakpoints = {
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+const keys = {
+  spacingUnit: '--DataGrid-t-spacing-unit',
+
+  /* Variables */
+  colors: {
+    border: {
+      base: '--DataGrid-t-colors-border-base',
+    },
+    foreground: {
+      base: '--DataGrid-t-colors-foreground-base',
+      muted: '--DataGrid-t-colors-foreground-muted',
+      accent: '--DataGrid-t-colors-foreground-accent',
+      disabled: '--DataGrid-t-colors-foreground-disabled',
+    },
+    background: {
+      base: '--DataGrid-t-colors-background-base',
+      overlay: '--DataGrid-t-colors-background-overlay',
+      backdrop: '--DataGrid-t-colors-background-backdrop',
+    },
+    interactive: {
+      hover: '--DataGrid-t-colors-interactive-hover',
+      hoverOpacity: '--DataGrid-t-colors-interactive-hover-opacity',
+      focus: '--DataGrid-t-colors-interactive-focus',
+      focusOpacity: '--DataGrid-t-colors-interactive-focus-opacity',
+      disabled: '--DataGrid-t-colors-interactive-disabled',
+      disabledOpacity: '--DataGrid-t-colors-interactive-disabled-opacity',
+      selected: '--DataGrid-t-colors-interactive-selected',
+      selectedOpacity: '--DataGrid-t-colors-interactive-selected-opacity',
+    },
+  },
+  cell: {
+    background: {
+      pinned: '--DataGrid-t-cell-background-pinned',
+    },
+  },
```

Comment:
We might need to add a header background color var as well if we want to support customizing it separately, see https://github.com/mui/mui-x/pull/16066/files#r1901714839

```diff
diff --git a/packages/x-data-grid/src/constants/cssVariables.ts b/packages/x-data-grid/src/constants/cssVariables.ts
index e63682820..cacf8f1d3 100644
--- a/packages/x-data-grid/src/constants/cssVariables.ts
+++ b/packages/x-data-grid/src/constants/cssVariables.ts
@@ -52,6 +52,11 @@ const keys = {
       pinned: '--DataGrid-t-cell-background-pinned',
     },
   },
+  header: {
+    background: {
+      base: '--DataGrid-t-header-background-base',
+    },
+  },
   radius: {
     base: '--DataGrid-t-radius-base',
   },
```

Reply:
Regardless, I can add it in that PR once this is merged.

Reply:
Sounds good, let's keep this as a refactor and add new features in separate PR:s.